utils/getAbi.ts

```diff
@@ -1,38 +1,95 @@
+  types: string[];
   abi: any[];
+  getState?: (instance: Contract) => Promise<any>;
 };
 export const Contracts: ContractType[] = [
   {
     // we could include semver to do compatibilty checks for version strings, but it would add to bundle and this is good enough
     versions: ["1", "1.2.0", "1.2.1", "1.2.2"],
-    names: ["EMP", "ExpiringMultiParty"],
+    types: ["EMP", "ExpiringMultiParty"],
     abi: emp1.abi,
+    async getState(instance: Contract) {
+      const state = {
+        expirationTimestamp: (await instance.expirationTimestamp()) as BigNumber,
+        collateralCurrency: (await instance.collateralCurrency()) as string, // address
+        priceIdentifier: (await instance.priceIdentifier()) as Bytes,
+        tokenCurrency: (await instance.tokenCurrency()) as string, // address
+        collateralRequirement: (await instance.collateralRequirement()) as BigNumber,
+        disputeBondPct: (await instance.disputeBondPct()) as BigNumber,
+        disputerDisputeRewardPct: (await instance.disputerDisputeRewardPct()) as BigNumber,
+        sponsorDisputeRewardPct: (await instance.sponsorDisputeRewardPct()) as BigNumber,
+        minSponsorTokens: (await instance.minSponsorTokens()) as BigNumber,
+        timerAddress: (await instance.timerAddress()) as string, // address
+        cumulativeFeeMultiplier: (await instance.cumulativeFeeMultiplier()) as BigNumber,
+        rawTotalPositionCollateral: (await instance.rawTotalPositionCollateral()) as BigNumber,
+        totalTokensOutstanding: (await instance.totalTokensOutstanding()) as BigNumber,
+        liquidationLiveness: (await instance.liquidationLiveness()) as BigNumber,
+        withdrawalLiveness: (await instance.withdrawalLiveness()) as BigNumber,
+        currentTime: (await instance.getCurrentTime()) as BigNumber,
+        contractState: Number(await instance.contractState()) as number,
+        finderAddress: (await instance.finder()) as string, // address
+        expiryPrice: (await instance.expiryPrice) as BigNumber,
+        isExpired: false,
+      };
+      state.isExpired = state.currentTime.gte(state.expirationTimestamp);
+      return state;
+    },
   },
   {
     versions: ["2", "2.0.0", "2.0.1", "latest"],
-    names: ["EMP", "ExpiringMultiParty"],
+    types: ["EMP", "ExpiringMultiParty"],
     abi: emp2.abi,
+    async getState(instance: Contract) {
+      const state = {
+        expirationTimestamp: (await instance.expirationTimestamp()) as BigNumber,
+        collateralCurrency: (await instance.collateralCurrency()) as string, // address
+        priceIdentifier: (await instance.priceIdentifier()) as Bytes,
+        tokenCurrency: (await instance.tokenCurrency()) as string, // address
+        collateralRequirement: (await instance.collateralRequirement()) as BigNumber,
+        disputeBondPct: (await instance.disputeBondPercentage()) as BigNumber,
+        disputerDisputeRewardPct: (await instance.disputerDisputeRewardPercentage()) as BigNumber,
```

Comment:
I wonder if you don't need to repeat the same calls for methods that are common across versions, and then just make the different calls per version:

const state = {
...commonState,
sponsorDisputeRewardPct: (await instance.sponsorDisputeRewardPercentage()) as BigNumber,
...
}

Reply:
yea that is a really good suggestion, there really is only a handful of functions